Update BFE, BT, CP, CE, GLXSB, EBUS, EMU, EMUJOY, HIFN, HME drivers to use PCI_MATCH

lakhanshiva opened the pull request.

lakhanshiva committed:
Update GLXSB to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update EBUS to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update EMU, EMUJOY to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update HIFN to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update BT to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update HME to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update CP to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update CE to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update ET to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva committed:
Update IPW to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

bsdimp commented on sys/dev/arcmsr/arcmsr.h:
```diff
@@ -1534,7 +1537,7 @@ struct SENSE_DATA {
 **                                                            In PCI-X mode this bit is also set when it forwards a SCM with a target abort error code.
 ** 10:09       01                             DEVSEL# Timing: Indicates slowest response to a non-configuration command on the primary interface. 
-**                                                            Returns ��01b�� when read, indicating that bridge responds no slower than with medium timing.
+**                                                            Returns ¡§01b¡¨ when read, indicating that bridge responds no slower than with medium timing.
```
what are all these changes?

bsdimp commented on sys/dev/buslogic/bt_pci.c:
```diff
@@ -56,16 +56,11 @@ __FBSDID("$FreeBSD$");
-	    "Buslogic Multi-Master SCSI Host Adapter"},
-	{0, 0, 0},
+struct pci_device_table bt_devs[] = {
+	{PCI_DEV(0x104B, 0x1040),
```
These should use symbolic stuff. Maybe I need a new macro PCI_DEVID :)

lakhanshiva replied:
Is this the macro formulation that we need here ?
#define PCI_DEVID(did)							\
	PCI_DEV((uint16_t)did, (uint16_t)(did >> 16))

bsdimp commented on sys/dev/buslogic/bt_pci.c:
```diff
@@ -56,16 +56,11 @@ __FBSDID("$FreeBSD$");
+struct pci_device_table bt_devs[] = {
+	{PCI_DEV(0x104B, 0x1040),
+	 PCI_DESCR("Buslogic Multi-Master SCSI Host Adapter")},
+	{PCI_DEV(0x104B, 0x0140),
```
And the number here should be 0x8130

bsdimp commented on sys/dev/ce/if_ce.c:
```diff
@@ -2636,11 +2636,9 @@ MODULE_DEPEND (ce, sppp, 1, 1, 1);
 DRIVER_MODULE (cemod, pci, ce_driver, ce_devclass, ce_modevent, NULL);
-MODULE_PNP_INFO("U16:vendor; U16:device;D:#", pci, cemod, ce_devs,
-     sizeof(ce_devs[0]), nitems(ce_devs));
+PCI_PNP_INFO(ce_devs);
```
where were ce_devs converted to the new pci_device_table ?

lakhanshiva replied:
It was converted earlier in.. https://github.com/bsdimp/freebsd/commit/991e00e955e1803e1c7e645bf8802351f089cf40

bsdimp commented on sys/dev/cp/if_cp.c:
```diff
@@ -207,11 +207,11 @@ static struct mbuf *makembuf (void *buf, unsigned len)
 	if (cpd == NULL)
 		return ENXIO;
-	device_set_desc(dev, cpd->description);
+	device_set_desc(dev, cpd->descr);
```
Same question, though maybe this was in a prior review.

lakhanshiva replied:
Yes, It was converted here.. https://github.com/bsdimp/freebsd/commit/f75f586106003d1671e4f57f951e719b3d6c51af

bsdimp commented on sys/dev/cy/cy_pci.c:
```diff
@@ -94,21 +92,24 @@ static driver_t cy_pci_driver = {
 	u_int32_t device_id;
 
 	device_id = pci_get_devid(dev);
 	device_id &= ~0x00060000;
```
This mask is troublesome. It means that we're looking at 4x the number of cards...

lakhanshiva replied:
Should we try and get rid of this mask then ?

lakhanshiva replied:
I am moving this to next PR so this PR can be merged.

bsdimp reviewed (commented):
These are pretty close (and may actually be right if the parts I'm asking about are in an earlier PR)

lakhanshiva committed:
Update BFE to use PCI_MATCH

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253

lakhanshiva commented:
On Sun, Jul 22, 2018 at 11:09 PM Warner Losh <notifications@github.com>
wrote:

> *@bsdimp* commented on this pull request.
> ------------------------------
>
> In sys/dev/arcmsr/arcmsr.h
> <https://github.com/bsdimp/freebsd/pull/9#discussion_r204248071>:
>
> > @@ -1534,7 +1537,7 @@ struct SENSE_DATA {
>  **									as the target of a transaction, terminates it with a Target Abort.
>  **                                                            In PCI-X mode this bit is also set when it forwards a SCM with a target abort error code.
>  ** 10:09       01                             DEVSEL# Timing: Indicates slowest response to a non-configuration command on the primary interface.
> -**                                                            Returns ��01b�� when read, indicating that bridge responds no slower than with medium timing.
> +**                                                            Returns ¡§01b¡¨ when read, indicating that bridge responds no slower than with medium timing.
>
> what are all these changes?
>
All this came in due to change in file ISO format. It is now fixed & moved
to PR #10

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bsdimp/freebsd/pull/9#pullrequestreview-139294459>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEiqmBFIn3sPHSAg70KN3gcWs008Rp4Kks5uJLjKgaJpZM4VIJMl>
> .
>

lakhanshiva commented:
On Sun, Jul 22, 2018 at 11:11 PM Warner Losh <notifications@github.com>
wrote:

> *@bsdimp* commented on this pull request.
> ------------------------------
>
> In sys/dev/buslogic/bt_pci.c
> <https://github.com/bsdimp/freebsd/pull/9#discussion_r204248111>:
>
> > @@ -56,16 +56,11 @@ __FBSDID("$FreeBSD$");
>  #define PCI_DEVICE_ID_BUSLOGIC_MULTIMASTER_NC	0x0140104Bul
>  #define PCI_DEVICE_ID_BUSLOGIC_FLASHPOINT	0x8130104Bul
>
> -static struct bt_dev {
> -	uint16_t vendorid;
> -	uint32_t deviceid;
> -	const char *description;
> -} bt_devs[] = {
> -	{0x104B, PCI_DEVICE_ID_BUSLOGIC_MULTIMASTER,
> -	    "Buslogic Multi-Master SCSI Host Adapter"},
> -	{0x104B, PCI_DEVICE_ID_BUSLOGIC_MULTIMASTER_NC,
> -	    "Buslogic Multi-Master SCSI Host Adapter"},
> -	{0, 0, 0},
> +struct pci_device_table bt_devs[] = {
> +	{PCI_DEV(0x104B, 0x1040),
>
> These should use symbolic stuff. Maybe I need a new macro PCI_DEVID :)
>
Yeah, Pl. let me know once it is done!

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bsdimp/freebsd/pull/9#pullrequestreview-139294499>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEiqmBWnTsY4v7IK7iN3_Nt_Q4b5rouCks5uJLkzgaJpZM4VIJMl>
> .
>

lakhanshiva commented:
On Sun, Jul 22, 2018 at 11:11 PM Warner Losh <notifications@github.com>
wrote:

> *@bsdimp* commented on this pull request.
> ------------------------------
>
> In sys/dev/buslogic/bt_pci.c
> <https://github.com/bsdimp/freebsd/pull/9#discussion_r204248126>:
>
> > @@ -56,16 +56,11 @@ __FBSDID("$FreeBSD$");
>  #define PCI_DEVICE_ID_BUSLOGIC_MULTIMASTER_NC	0x0140104Bul
>  #define PCI_DEVICE_ID_BUSLOGIC_FLASHPOINT	0x8130104Bul
>
> -static struct bt_dev {
> -	uint16_t vendorid;
> -	uint32_t deviceid;
> -	const char *description;
> -} bt_devs[] = {
> -	{0x104B, PCI_DEVICE_ID_BUSLOGIC_MULTIMASTER,
> -	    "Buslogic Multi-Master SCSI Host Adapter"},
> -	{0x104B, PCI_DEVICE_ID_BUSLOGIC_MULTIMASTER_NC,
> -	    "Buslogic Multi-Master SCSI Host Adapter"},
> -	{0, 0, 0},
> +struct pci_device_table bt_devs[] = {
> +	{PCI_DEV(0x104B, 0x1040),
> +	 PCI_DESCR("Buslogic Multi-Master SCSI Host Adapter")},
> +	{PCI_DEV(0x104B, 0x0140),
>
> And the number here should be 0x8130
>
Corrected and updated this!

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bsdimp/freebsd/pull/9#pullrequestreview-139294513>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEiqmKeWpwWtOn_Dp-2Wuy8cmPVB0OEbks5uJLlSgaJpZM4VIJMl>
> .
>

lakhanshiva commented:
On Sun, Jul 22, 2018 at 11:14 PM Warner Losh <notifications@github.com>
wrote:

> *@bsdimp* commented on this pull request.
> ------------------------------
>
> In sys/dev/cy/cy_pci.c
> <https://github.com/bsdimp/freebsd/pull/9#discussion_r204248193>:
>
> >  MODULE_DEPEND(cy, pci, 1, 1, 1);
>
>  static int
>  cy_pci_probe(dev)
>  	device_t dev;
>  {
> +	const struct pci_device_table *cyd;
>  	u_int32_t device_id;
>
>  	device_id = pci_get_devid(dev);
>  	device_id &= ~0x00060000;
>
> This mask is troublesome. It means that we're looking at 4x the number of
> cards...
>
I will try to get rid of this mask ?

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bsdimp/freebsd/pull/9#pullrequestreview-139294580>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEiqmKtApp4TDOUyAFNzQCrtD0vrlB3fks5uJLnggaJpZM4VIJMl>
> .
>

lakhanshiva committed:
Add PCI_DEVID macro 

Reviewed by: imp, chuck
Submitted by: Lakhan Shiva Kamireddy <lakhanshiva@gmail.com>
Sponsored by: Google, Inc. (GSoC 2018)
Pull Request: https://github.com/bsdimp/freebsd/pull/9
Url: https://reviews.freebsd.org/D16253
